### ISSUE TITLE
[deploy] `klib` packaging is treated as error

### ISSUE THREAD

**Issue by sanyavertolet** [jreleaser/jreleaser#1746]
I develop Kotlin Multiplatform library and I want to release it with `JReleaser`. I set up `JReleaser` correctly (jreleaserConfig works ok). Here is the error I get:
```
[INFO]    [maven] Deploying all staged artifacts
[INFO]      [mavenCentral] Deploying to sonatype
[INFO]      [mavenCentral] Verifying prerequesites
[INFO]      [mavenCentral] Verifying POMs

[ERROR]     [mavenCentral] java.lang.IllegalStateException: org.apache.maven.project.ProjectBuildingException: Some problems were encountered while processing the POMs:
[ERROR]     [mavenCentral] [ERROR] Unknown packaging: klib @ line 12, column 14
```
I guess the problem is that kotlin multiplatform libs have `klib` packaging, which is treated as error by `pomchecker`. 

I would love to use `JReleaser` in order to release my Kotlin Multiplatform library, which is assembled and distributed as `.klib` package. In order to do so, pomchecker needs to be modified.

**Comment by aalmiray** [comment]
No. Pomchecker needs no updates. A full Maven model needs to be resolved which means a plugin must be configured in the pom.xml to handle the `klib` packaging type.

**Comment by aalmiray** [comment]
See https://github.com/jreleaser/jreleaser/discussions/1705 for additional details.

**Comment by sanyavertolet** [comment]
Thanks, that helped, but now I get another error:
```
[INFO]  Deploying Maven artifacts
[INFO]    [maven] Deploying all staged artifacts
[INFO]      [mavenCentral] Deploying to sonatype
[INFO]      [mavenCentral] Verifying prerequesites
[ERROR]     [mavenCentral] cookie-js-0.1.0-20241016.143512-2.jar is missing
```

Seems that the error is produced by `Maven Central` but not `JReleaser`, still you might know the solution.
Maybe there is a way to explicitly tell that kotlin multiplatform package is being released so that there is no use to look for `.jar` artifact?

Here is my `staging-deploy`:
```
└── io
    └── github
        └── sanyavertolet
            └── jswrappers
                └── cookie-js
                    ├── 0.1.0-SNAPSHOT
                    │   ├── cookie-js-0.1.0-20241016.143512-1-javadoc.jar
                    │   ├── cookie-js-0.1.0-20241016.143512-1-javadoc.jar.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-1-javadoc.jar.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-1-javadoc.jar.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-1-javadoc.jar.sha512
                    │   ├── cookie-js-0.1.0-20241016.143512-1-kotlin-tooling-metadata.json
                    │   ├── cookie-js-0.1.0-20241016.143512-1-kotlin-tooling-metadata.json.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-1-kotlin-tooling-metadata.json.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-1-kotlin-tooling-metadata.json.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-1-kotlin-tooling-metadata.json.sha512
                    │   ├── cookie-js-0.1.0-20241016.143512-1-sources.jar
                    │   ├── cookie-js-0.1.0-20241016.143512-1-sources.jar.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-1-sources.jar.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-1-sources.jar.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-1-sources.jar.sha512
                    │   ├── cookie-js-0.1.0-20241016.143512-1.jar
                    │   ├── cookie-js-0.1.0-20241016.143512-1.jar.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-1.jar.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-1.jar.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-1.jar.sha512
                    │   ├── cookie-js-0.1.0-20241016.143512-1.module
                    │   ├── cookie-js-0.1.0-20241016.143512-1.module.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-1.module.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-1.module.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-1.module.sha512
                    │   ├── cookie-js-0.1.0-20241016.143512-1.pom
                    │   ├── cookie-js-0.1.0-20241016.143512-1.pom.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-1.pom.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-1.pom.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-1.pom.sha512
                    │   ├── cookie-js-0.1.0-20241016.143512-2-javadoc.jar
                    │   ├── cookie-js-0.1.0-20241016.143512-2-javadoc.jar.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-2-javadoc.jar.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-2-javadoc.jar.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-2-javadoc.jar.sha512
                    │   ├── cookie-js-0.1.0-20241016.143512-2-sources.jar
                    │   ├── cookie-js-0.1.0-20241016.143512-2-sources.jar.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-2-sources.jar.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-2-sources.jar.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-2-sources.jar.sha512
                    │   ├── cookie-js-0.1.0-20241016.143512-2.klib
                    │   ├── cookie-js-0.1.0-20241016.143512-2.klib.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-2.klib.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-2.klib.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-2.klib.sha512
                    │   ├── cookie-js-0.1.0-20241016.143512-2.module
                    │   ├── cookie-js-0.1.0-20241016.143512-2.module.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-2.module.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-2.module.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-2.module.sha512
                    │   ├── cookie-js-0.1.0-20241016.143512-2.pom
                    │   ├── cookie-js-0.1.0-20241016.143512-2.pom.md5
                    │   ├── cookie-js-0.1.0-20241016.143512-2.pom.sha1
                    │   ├── cookie-js-0.1.0-20241016.143512-2.pom.sha256
                    │   ├── cookie-js-0.1.0-20241016.143512-2.pom.sha512
                    │   ├── maven-metadata.xml
                    │   ├── maven-metadata.xml.md5
                    │   ├── maven-metadata.xml.sha1
                    │   ├── maven-metadata.xml.sha256
                    │   └── maven-metadata.xml.sha512
                    ├── maven-metadata.xml
                    ├── maven-metadata.xml.md5
                    ├── maven-metadata.xml.sha1
                    ├── maven-metadata.xml.sha256
                    └── maven-metadata.xml.sha512
```

**Comment by sanyavertolet** [comment]
Seems that @ygdrasil-io was able to post `.klib` packages to `Central Portal` so it seems to be possible. It would be nice to use `JReleaser` for Kotlin releases as well as it provides a nice way to configure all the release steps, but now there is no documentation on Kotlin releases.

**Comment by aalmiray** [comment]
In this case there are 2 things that can be added to JReleaser:

1. Relax validation with another `artifactOverride` setting that skips checking for explicit `.jar` matching a given `.pom`.
2. Explicit check for `.klib` artifact matching a given `.pom` with `klib` in its `<packaging>`.

The first allows for any packaging to be used without forcing a companion `.jar` artifact. This serves as a escape hatch for any "unsupported" packaging setting.
The second will check explicitly for `.klib`.

**Comment by aalmiray** [comment]
FWIW https://mvnrepository.com/artifact/com.soywiz.korge/korge/6.0.0-beta4 does not publish a pom with `<packaging>klib</packaging>` nor an artifact with `.klib` extension.

You have two pom files with the same `artifactId`/`version` combination which look different at the moment due to the version being snapshot (`cookie-js-0.1.0-20241016.143512-1.pom`, `cookie-js-0.1.0-20241016.143512-2.pom`). You'll get in trouble when publishing no snapshot releases. One of these poms must have a different filename, either by choosing a different `artifactId` or adding a `classifier`.

**Comment by aalmiray** [comment]
The ability to skip checking for a companion JAR using `artifactOverrides` is now in place.
Additional checks for companion artifacts matching a given packaging is still forthcoming.

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.15.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.15.0))
